### ISSUE TITLE
pyright not propery detecting modules when using a virtual environment

### ISSUE THREAD

**Issue by Jojobinx17** [zed-industries/zed#18469]
### Check for existing issues

- [x] Completed

### Describe the bug / provide steps to reproduce it

Pyright does not properly acknowledge modules in a python virtual environment. 

After following the [docs](https://zed.dev/docs/languages/python#virtual-environments), I have a virtual environment that runs perfectly and works as intended. However, Pyright seems to think that a bunch of all of my modules' functions don't exist! 

I have all my modules in a folder called 'venv', located in my project's root directory.

With no `pyrightconfig.json` file set up, pyright rightfully displays errors on lines like `from datetime import datetime`, as it doesn't know where all my modules are.

However, when I create a `pyrightconfig.json` file in my project's root directory...
```
{
  "venvPath": ".",
  "venv": "venv"
}
```

...pyright switches up. It finds my modules (the error on `from datetime import datetime` goes away, woo!), but I get _new_ errors when trying to access my modules' functions:

`e_date = s_date + datetime.timedelta(days=30)` gives -> _Pyright: Cannot access attribute "timedelta" for class "type[datetime]". Attribute "timedelta" is unknown_.

This isn't an error with my code, everything works perfectly when tried in vscode (no errors are detected), and the project runs perfectly when tested. 

In conclusion:
Pyright finds my modules properly, but can't seem to find their functions when I import them.

### Environment

Zed: v0.154.3 (Zed)
OS: macOS 14.6.0
Memory: 16 GiB
Architecture: aarch64

**Comment by notpeter** [comment]
Would you be willing to provide a create a minimal repo and steps to reproduce this?

I wonder if it is related this similar issue where pyright chokes on resolving attributes which have been decorated:
- https://github.com/zed-industries/zed/issues/14371

**Comment by Jojobinx17** [comment]
Okay, creating a minimal example required me to do a little more digging on the root of the issue, and I think I've figured it out what's going on.

Here's what I ended up with:

```python
from django.db import models

def test():
    print(Person.objects.all())
    return

class Person(models.Model):
    first_name = models.CharField(max_length=30)
    last_name = models.CharField(max_length=30)
```

 `Person.objects.all()` is wrongfully giving:
```
Pyright: Cannot access attribute "objects" for class "type[Person]"
 Attribute "objects" is unknown
```

I'm pretty sure the error is _actually related to Django [managers](https://docs.djangoproject.com/en/5.1/topics/db/managers/#managers)_, seen in the example above. As it turns out, all of the errors that appeared in my codebase are directly related to using them. ☹️

When I made the previous example, I just picked one of the 200 errors that I was getting, and as it turns out, it was completely unrelated to the root issue (`datetime` wasn't actually broken, it was _another line using a manager that was breaking it_). Oops. 

If you have any thoughts (or next steps), please let me know!

**Comment by notpeter** [comment]
Hi! 

I was able to reproduce your issue with the following steps:

```
mkdir -p ~/source/issue18469
cd $_
uv init
uv venv
uv pip install django
zed . hello.py
```

Paste contents of the file above into hello.py and I see the same error:

> Pyright: Cannot access attribute "objects" for class "type[Person]" Attribute "objects" is unknown
<img width="661" alt="Screenshot 2024-11-19 at 10 25 12" src="https://github.com/user-attachments/assets/6b16bc58-0489-4acc-9fe9-614dd797b9d1">

----

Solution:
```
uv add --dev 'django-stubs[compatible-mypy]' django-types
```

<img width="1013" alt="Screenshot 2024-11-19 at 10 46 15" src="https://github.com/user-attachments/assets/ddcf6abf-a940-4e2c-87dc-7f294edb3523">

Can you see if adding django-types and django-stubs to your venv fixes things?

**Comment by notpeter** [comment]
Here's a link to repro repo: 
- https://github.com/notpeter/issue18469/

Which triggers this:
![image](https://github.com/user-attachments/assets/7110902d-7816-4a97-b974-e37d48df4134)

**Comment by Jojobinx17** [comment]
> Can you see if adding django-types and django-stubs to your venv fixes things?

Yup, this worked. Thank you so much!
